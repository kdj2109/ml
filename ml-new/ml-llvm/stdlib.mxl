void printFloatTuples(float[] x,  int len) {
         int i;
         float f;
         for (i = 0; i < len; i = i + 1) {
                 f = $x;
		 printf(f);
		 x = x.+;
         }
}

void printIntTuples(int[] x,  int len) {
         int i;
         int f;
         for (i = 0; i < len; i = i + 1) {
                 f = $x;
		 print(f);
		 x = x.+;
         }
}

void addIntTuples(int[] x, int[] y, int len) {
	/*

	Arguments x and y are integer pointers
	pointing to the first elements in the tuples.
	Length is the length of the tuples being passed in.
	Adds the second tuple to the first tuple, stores in the first
	*/
	int i;

	for (i = 0; i < len; i = i + 1) {
		$x = $x + $y; // $ dereferences the pointer
		x = x.+;      // Incrementing the pointer by 1
		y = y.+;		  // Incrementing the pointer by 1
	}
}
