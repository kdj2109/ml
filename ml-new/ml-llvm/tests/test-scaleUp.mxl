#include <stdlib.mxl>;
void scaleUp (int[][][] x, int[][][] y, int rX, int cX, int rY, int cY, int l){
  int rFactor;
  int cFactor;
  int a;
  int b;
  int i;
  int k;
  int j;
  int[][][] pointer;
  int[][][] startOfTuple;

  rFactor=rY/rX;
  cFactor=cY/cX;


  for (a=0; a<rX; a=a+1){
    for (i=0; i<rFactor; i=i+1){
    //create a new pointer starting at first column at this row of i every time I think maybe
      startOfTuple=x;
      for (b=0; b<cX; b=b+1){
        for (j=0; j<cFactor; j=j+1){
          pointer=startOfTuple;
          for (k=0; k<l; k=k+1){
            $y=$pointer;
            y=y.+;
            pointer=pointer.+;
          }
        }
        for (k=0; k<l; k=k+1){
          startOfTuple=startOfTuple.+;
        } 
      }
    }
    x=pointer;
  }


} 



int main(){
    int[3][5:2] mat;
    int[3][10:4] y;
    mat=[|(1,2,3),(1,2,3)|(200,10,20),(1,2,3)| (200,10,20),(1,77,3) |(200,10,20),(1,88,3) |(200,10,20),(1,2,99)|];



  scaleUp(@@@mat,@@@y,mat.rows,mat.columns,y.rows,y.columns,3);
  out(@@@y, y.rows, y.columns, y.length);

    return 0;
}