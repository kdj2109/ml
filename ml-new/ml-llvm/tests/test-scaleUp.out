P3
10 4
255
1 2 3 1 2 3 1 2 3 1 2 3 
1 2 3 1 2 3 1 2 3 1 2 3 
200 10 20 200 10 20 1 2 3 1 2 3 
200 10 20 200 10 20 1 2 3 1 2 3 
200 10 20 200 10 20 1 77 3 1 77 3 
200 10 20 200 10 20 1 77 3 1 77 3 
200 10 20 200 10 20 1 88 3 1 88 3 
200 10 20 200 10 20 1 88 3 1 88 3 
200 10 20 200 10 20 1 2 99 1 2 99 
200 10 20 200 10 20 1 2 99 1 2 99 